components/ToggleMode.tsx

```diff
@@ -1,19 +1,37 @@
 "use client";
 import { ThemeContext } from "@/context/ThemeContext";
-import { useContext, useEffect, useState } from "react";
+import { useContext, useEffect, useRef, useState } from "react";
+import { FaArrowCircleUp } from "react-icons/fa";
 
 export default function ToggleMood() {
   const { changeTheme }: any = useContext(ThemeContext);
   const [theme, setTheme] = useState<string>("winter");
+  const [scrollTop, setScrollTop] = useState<boolean>(false);
+  const scrollUpRef = useRef<HTMLDivElement>(null);
+
   useEffect(() => {
     changeTheme(theme);
   }, [theme, changeTheme]);
   const toggleMode = function () {
     setTheme(theme === "dark" ? "winter" : "dark");
     changeTheme(theme === "dark" ? "winter" : "dark");
   };
+  useEffect(() => {
+    const handleScrollUp = () => {
+      const { scrollTop } = document.documentElement;
+      scrollTop >= 1200 ? setScrollTop(true) : setScrollTop(false);
```

Comment:
I am wondering why exactly 1200? If this is a screen resolution, let’s put this into some file that includes only constant values across the codebase and import it from that file here.